08-components-lab/src/03-colorpicker.jsx

```diff
@@ -1,17 +1,48 @@
- * Implement a color picker component.
- *
- * Component should show an <input> field to select a color, and below a <div>
- * When a color is selected, that color should be used as the background color
- * of the <div>
- *
- * Bonus: Allow external code to pass some text to be used as the contents of the
- * colored <div>
- */
-
-import React from 'react';
-
-export default React.createClass({
-});
+/**
+ * Created by liat_be on 18/12/2016.
+ */
+import React from 'react';
+
+export default class ColorPicker extends React.Component {
+  //  CSS_COLOR_NAMES = ["AliceBlue","AntiqueWhite","Aqua","Aquamarine","Azure","Beige","Bisque","Black","BlanchedAlmond","Blue","BlueViolet","Brown","BurlyWood","CadetBlue","Chartreuse","Chocolate","Coral","CornflowerBlue","Cornsilk","Crimson","Cyan","DarkBlue","DarkCyan","DarkGoldenRod","DarkGray","DarkGrey","DarkGreen","DarkKhaki","DarkMagenta","DarkOliveGreen","Darkorange","DarkOrchid","DarkRed","DarkSalmon","DarkSeaGreen","DarkSlateBlue","DarkSlateGray","DarkSlateGrey","DarkTurquoise","DarkViolet","DeepPink","DeepSkyBlue","DimGray","DimGrey","DodgerBlue","FireBrick","FloralWhite","ForestGreen","Fuchsia","Gainsboro","GhostWhite","Gold","GoldenRod","Gray","Grey","Green","GreenYellow","HoneyDew","HotPink","IndianRed","Indigo","Ivory","Khaki","Lavender","LavenderBlush","LawnGreen","LemonChiffon","LightBlue","LightCoral","LightCyan","LightGoldenRodYellow","LightGray","LightGrey","LightGreen","LightPink","LightSalmon","LightSeaGreen","LightSkyBlue","LightSlateGray","LightSlateGrey","LightSteelBlue","LightYellow","Lime","LimeGreen","Linen","Magenta","Maroon","MediumAquaMarine","MediumBlue","MediumOrchid","MediumPurple","MediumSeaGreen","MediumSlateBlue","MediumSpringGreen","MediumTurquoise","MediumVioletRed","MidnightBlue","MintCream","MistyRose","Moccasin","NavajoWhite","Navy","OldLace","Olive","OliveDrab","Orange","OrangeRed","Orchid","PaleGoldenRod","PaleGreen","PaleTurquoise","PaleVioletRed","PapayaWhip","PeachPuff","Peru","Pink","Plum","PowderBlue","Purple","Red","RosyBrown","RoyalBlue","SaddleBrown","Salmon","SandyBrown","SeaGreen","SeaShell","Sienna","Silver","SkyBlue","SlateBlue","SlateGray","SlateGrey","Snow","SpringGreen","SteelBlue","Tan","Teal","Thistle","Tomato","Turquoise","Violet","Wheat","White","WhiteSmoke","Yellow","YellowGreen"];
+
+    constructor(props) {
+        super(props);
+
+        const style = {
+          margin: "20px",
+          height:"100px",
+          width: "100px",
+          backgroundColor: "none"
+        };
+
+        this.state = {
+            style
+        }
+
+
+    }
+
+    handleChange(e) {
+        const color = e.target.value;
+        const newStyle = Object.assign({}, this.state.style, {backgroundColor:color});
+
+        this.setState({style: newStyle});
+    }
+
+    render() {
+        return (
+            <div>
+                <input type="color" onChange={this.handleChange.bind(this)}/>
+               {/* <select onChange={this.handleChange.bind(this)}>

```

Comment:
Look previous comment on binding. Best is to use arrow functions or binding constructor